### PR TITLE
Added doxyfiles

### PR DESCRIPTION
**Pull request by ahcorde**
Added Doxyfile to `rmw_fastrtps_cpp`, `rmw_fastrtps_dynamic_cpp` and `rmw_fastrtps_shared_cpp` in the process of reaching Quality Level 1.

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rmw_fastrtps_cpp/Doxyfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+EXTRACT_ALL            = YES
+SORT_MEMBER_DOCS       = NO
+
+GENERATE_LATEX         = NO
+
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             = RCLCPP_PUBLIC=
```
Comment (brawner) [inline rmw_fastrtps_cpp/Doxyfile:24]: These use their own visibility control macros. Fix here and below.

File: rmw_fastrtps_cpp/Doxyfile [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# All settings not listed here will use the Doxygen default values.
+
+PROJECT_NAME           = "rmw_fastrtps_cpp"
+PROJECT_NUMBER         = master
+PROJECT_BRIEF          = "Package containing tools for dynamically loadable components"
```
Comment (brawner) [inline rmw_fastrtps_cpp/Doxyfile:5]: Wrong package description.

File: rmw_fastrtps_cpp/Doxyfile [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+PREDEFINED             = RCLCPP_PUBLIC=
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
+# Consider changing "latest" to the version you want to reference (e.g. beta1 or 1.0.0)
+TAGFILES += "../../../../doxygen_tag_files/rclcpp.tag=http://docs.ros2.org//api/rclcpp/"
+TAGFILES += "../../../../doxygen_tag_files/rcl_lifecycle.tag=http://docs.ros2.org//api/rcl_lifecycle/"
+TAGFILES += "../../../../doxygen_tag_files/rcl.tag=http://docs.ros2.org//api/rcl/"
+TAGFILES += "../../../../doxygen_tag_files/rmw.tag=http://docs.ros2.org//api/rmw/"
+TAGFILES += "../../../../doxygen_tag_files/rcutils.tag=http://docs.ros2.org//api/rcutils/"
```
Comment (brawner) [inline rmw_fastrtps_cpp/Doxyfile:33]: You might also want to link to rmw_fastrtps_shared_cpp.

File: rmw_fastrtps_cpp/Doxyfile [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             = RCLCPP_PUBLIC=
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
+# Consider changing "latest" to the version you want to reference (e.g. beta1 or 1.0.0)
+TAGFILES += "../../../../doxygen_tag_files/rclcpp.tag=http://docs.ros2.org//api/rclcpp/"
+TAGFILES += "../../../../doxygen_tag_files/rcl_lifecycle.tag=http://docs.ros2.org//api/rcl_lifecycle/"
```
Comment (brawner) [inline rmw_fastrtps_cpp/Doxyfile:30]: Remove

File: rmw_fastrtps_dynamic_cpp/Doxyfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# All settings not listed here will use the Doxygen default values.
+
+PROJECT_NAME           = "rmw_fastrtps_dynamic_cpp"
+PROJECT_NUMBER         = master
+PROJECT_BRIEF          = "Implement the ROS middleware interface using eProsima FastRTPS  static code generation in C++."
```
Comment (brawner) [inline rmw_fastrtps_dynamic_cpp/Doxyfile:5]: It looks like the package.xml is incorrect, but this should be "dynamic code generation"

File: rmw_fastrtps_dynamic_cpp/Doxyfile [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             = RMW_FASTRTPS_DYNAMIC_CPP_PUBLIC=
+
+# Tag files that do not exist will produce a warning and cross-project linking will not work.
+TAGFILES += "../../../../doxygen_tag_files/cppreference-doxygen-web.tag.xml=http://en.cppreference.com/w/"
+# Consider changing "latest" to the version you want to reference (e.g. beta1 or 1.0.0)
+TAGFILES += "../../../../doxygen_tag_files/rcl.tag=http://docs.ros2.org//api/rcl/"
+TAGFILES += "../../../../doxygen_tag_files/rmw.tag=http://docs.ros2.org//api/rmw/"
+TAGFILES += "../../../../doxygen_tag_files/rcutils.tag=http://docs.ros2.org//api/rcutils/"
```
Comment (brawner) [inline rmw_fastrtps_dynamic_cpp/Doxyfile:29]: Also link to rmw_fastrtps_shared_cpp

File: rmw_fastrtps_shared_cpp/Doxyfile [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+EXTRACT_ALL            = YES
+SORT_MEMBER_DOCS       = NO
+
+GENERATE_LATEX         = NO
+
+ENABLE_PREPROCESSING   = YES
+MACRO_EXPANSION        = YES
+EXPAND_ONLY_PREDEF     = YES
+PREDEFINED             = RMW_FASTRTPS_DYNAMIC_CPP_PUBLIC=
```
Comment (brawner) [inline rmw_fastrtps_shared_cpp/Doxyfile:22]: Change to SHARED

**Comment by ahcorde** [comment]
Added `/latest/` in the URL because this work will be replace by the `RELEASE_NAME` 

https://github.com/ros2/docs.ros2.org/blob/doc_gen/build_docs.sh#L30

File: rmw_fastrtps_cpp/Doxyfile [hunk 8]
```diff
@@ -0,0 +1,35 @@
+# All settings not listed here will use the Doxygen default values.
+
+PROJECT_NAME           = "rmw_fastrtps_cpp"
+PROJECT_NUMBER         = master
+PROJECT_BRIEF          = "Implement the ROS middleware interface using eProsima FastRTPS static code generation in C++."
+
+# Use these lines to include the generated logging.hpp (update install path if needed)
```
Comment (wjwwood) [inline rmw_fastrtps_cpp/Doxyfile:7]: This looks like a copy-paste error.

Comment (ahcorde) [reply rmw_fastrtps_cpp/Doxyfile]: https://github.com/ros2/rmw_fastrtps/pull/372/commits/1f7761e2cf2a0f9117bf0c81b6ad9ea80cb1c0b8

**Pull request merged**